src/storage/mod.rs

```diff
@@ -19,22 +31,128 @@ pub trait Engine {
 /// table (like column names, column types, storage engine, ...). It's `access`
 /// method locks the table globally and returns a storage engine to access
 /// the table data.
-pub struct Table;
+
+pub enum DatabaseError{
+    Io(Error),
+    Bin(EncodingError),
+}
+
+impl From<io::Error> for DatabaseError {
+    fn from(err: io::Error) -> DatabaseError {
+        DatabaseError::Io(err)
+    }
+}
+
+impl  From<EncodingError> for DatabaseError {
+    fn from(err: EncodingError) -> DatabaseError {
+        DatabaseError::Bin(err)
+    }
+}
+
+/// A Enum for File Modes
+///
+/// Can be used to define the save and load configuration of open_file
+#[derive(Debug)]
+enum FileMode{LoadDefault, SaveDefault,}
+
+/// A Enum for Datatypes (will be removed later)
+#[repr(u8)]
+#[derive(Clone,Copy,Debug,RustcDecodable, RustcEncodable)]
+enum DataType{ Integer = 1, Float = 2,}
+impl DataType{
+    pub fn value(&self) -> u8{
+       *self as u8
+    }
+}
+
+pub struct Database {
+    name: String,
+}
+
+impl Database {
+    pub fn new_database(database_name: &str) -> Database{
+        Database{name: database_name.to_string()}
+    }
+    pub fn create_database(&self) -> Result<(),DatabaseError>{
+        println!("trying to create dir!");
+        try!(create_dir(&self.name));
+        println!("created dir");
+        Ok(())
+    }
+}
+
+#[derive(Debug,RustcDecodable, RustcEncodable)]
+pub struct Table {
+    engine_id: u8,
+    version_nmbr: u8,
+    magic_nmbr: u8,
+    column_nmbr: u16,
+    columns: Vec<Column>,
+
+}
 
 impl Table {
-    pub fn load(database: &str, table: &str) -> Table {
+    pub fn load(database: &str, table: &str) -> Result<(), DatabaseError> {
         // TODO: Read the .tbl file from disk and parse it
-        Table
+        let mut file = try!(Self::open_file(database,table,FileMode::LoadDefault));
+
+        let data: Table = decode_from(&mut file, SizeLimit::Infinite).unwrap();
+        println!("{:?}", data);
+        Ok(())
+    }
+
+    pub fn create_new() -> Table {
```

Comment:
I assume this function is currently just for testing purposes... But some when you will provide a function to create a new `Table` object. This function should either be `fn new(name: &str, engine_id: u8)` or -- if you want to provide a blanco object -- you should implement the `Default` trait from `std`.